renku/domain_model/dataset.py

```diff
@@ -664,14 +673,9 @@ class ImageObjectRequestJson(marshmallow.Schema):
     mirror_locally = marshmallow.fields.Bool(default=False)
 
 
-def get_dataset_data_dir(client, dataset_name: str) -> str:
-    """Return default data directory for a dataset."""
-    return os.path.join(client.data_dir, dataset_name)
-
-
 def get_file_path_in_dataset(client, dataset: Dataset, dataset_file: DatasetFile) -> Path:
     """Return path of a file relative to dataset's data dir."""
     try:
-        return (client.path / dataset_file.entity.path).relative_to(get_dataset_data_dir(client, dataset.name))
+        return (client.path / dataset_file.entity.path).relative_to(dataset.get_datadir(client))
```

Comment:
I've just realized this function won't work correctly (no unit tests 😭) since `get_dataset_data_dir` doesn't return an absolute path. It should be like:
```
return (client.path / dataset_file.entity.path).relative_to(client.path / dataset.get_datadir(client))
```